From e53a1816aee84e4a56a9eb6c9543cc8c07e2de57 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Fri, 29 May 2015 15:13:21 +0300
Subject: [PATCH] drm: rcar-du: parse dt adv7511 i2c address

Parse DT for adv7511 i2c address for RCar-DU.
ADV7511 may have 2 different i2c address.

signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_encoder.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_encoder.c b/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
index de0ccc1..aff3a74 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_encoder.c
@@ -187,6 +187,7 @@ int rcar_du_encoder_init(struct rcar_du_device *rcdu,
 	struct rcar_du_encoder *renc;
 	unsigned int encoder_type;
 	int ret;
+	u32 val;
 	struct i2c_adapter *adapter;
 	struct i2c_board_info info[] = {
 		{
@@ -209,6 +210,9 @@ int rcar_du_encoder_init(struct rcar_du_device *rcdu,
 
 	info->of_node = of_find_node_by_name(NULL, "adv7511");
 
+	if (of_property_read_u32(info->of_node, "adi,reg", &val) == 0)
+		info->addr = val;
+
 	renc = devm_kzalloc(rcdu->dev, sizeof(*renc), GFP_KERNEL);
 	if (renc == NULL)
 		return -ENOMEM;
-- 
1.9.1

